test/org/sagebionetworks/bridge/dynamodb/DynamoTaskDaoMockTest.java

```diff
@@ -33,110 +41,111 @@
-        
+
         // This is the part that will need to be expanded per test.
         mapper = mock(DynamoDBMapper.class);
-        when(mapper.query(
-            (Class<DynamoTask>)any(Class.class), 
-            (DynamoDBQueryExpression<DynamoTask>)any(DynamoDBQueryExpression.class)
-        )).thenReturn(null);
-        
+        when(
+                        mapper.query((Class<DynamoTask>) any(Class.class),
+                                        (DynamoDBQueryExpression<DynamoTask>) any(DynamoDBQueryExpression.class)))
+                        .thenReturn(null);
+
         taskDao = new DynamoTaskDao();
         taskDao.setDdbMapper(mapper);
     }
-    
+
     @After
     public void after() {
         DateTimeUtils.setCurrentMillisSystem();
     }
-    
+
     @SuppressWarnings("unchecked")
     private void mockQuery(final List<Task> tasks) {
-        List<DynamoTask> results = Lists.newArrayList();
-        if (tasks != null) {
-            for (final Task task : tasks) {
-                results.add((DynamoTask)task);
-            }
-            when(mapper.load(any())).thenAnswer(new Answer<DynamoTask>() {
-                @Override public DynamoTask answer(InvocationOnMock invocation) throws Throwable {
-                    DynamoTask thisTask = (DynamoTask)invocation.getArguments()[0];
-                    for (Task task : tasks) {
-                        if (thisTask.getGuid().equals(task.getGuid()) && 
-                            thisTask.getHealthCode().equals(task.getHealthCode())) {
-                            return thisTask;
-                        }
+        when(mapper.load(any())).thenAnswer(new Answer<DynamoTask>() {
+            @Override
+            public DynamoTask answer(InvocationOnMock invocation) throws Throwable {
+                DynamoTask thisTask = (DynamoTask) invocation.getArguments()[0];
+                for (Task task : tasks) {
+                    if (thisTask.getGuid().equals(task.getGuid())
+                                    && thisTask.getHealthCode().equals(task.getHealthCode())) {
+                        return thisTask;
                     }
-                    return null;
                 }
-                
-            });
-        }
-        final PaginatedQueryList<DynamoTask> queryResults = (PaginatedQueryList<DynamoTask>)mock(PaginatedQueryList.class);
-        when(queryResults.iterator()).thenReturn(results.iterator());
-        when(queryResults.toArray()).thenReturn(results.toArray());
-        when(mapper.query(
-            (Class<DynamoTask>)any(Class.class), 
-            (DynamoDBQueryExpression<DynamoTask>)any(DynamoDBQueryExpression.class)
-        )).thenReturn(queryResults);
+                return null;
+            }
+
+        });
+        final PaginatedQueryList<DynamoTask> queryResults = (PaginatedQueryList<DynamoTask>) mock(PaginatedQueryList.class);
+        when(queryResults.iterator()).thenReturn(((List<DynamoTask>)(List<?>)tasks).iterator());
```

Comment:
You can't case from List<Task> to List<DynamoTask>, unless you effectively erase the generic with List<?> first. I don't pretend to understand the details, Eclipse doesn't build unless I do this or something complicated like copying tasks from one list to another.